omigami/feature_selector.py

```diff
@@ -22,68 +37,198 @@ def __init__(
+    ) -> OuterLoopResults:
+
+        raw_feature_elim_results = {}
+        feature_set = list(range(input_data.n_features))
+
+        while len(feature_set) >= self._minimum_features:
+            inner_results = []
+
+            for inner_split in data_splitter.iter_inner_splits(outer_split):
+                inner_loop_data = data_splitter.split_data(
+                    input_data, inner_split, feature_set
+                )
+
+                feature_evaluation_results = self.feature_evaluator.evaluate_features(
+                    inner_loop_data, feature_set
+                )
+
+                inner_results.append(feature_evaluation_results)
+
+            raw_feature_elim_results[tuple(feature_set)] = inner_results
+            feature_set = self._remove_features(feature_set, inner_results)
+
+        feature_elimination_results = self.post_processor.process_feature_elim_results(
+            raw_feature_elim_results
+        )
+        outer_loop_results = self.create_outer_loop_results(
+            feature_elimination_results, input_data, outer_split, data_splitter
         )
 
-    def _make_outer_loop(self, input_data: InputData) -> OuterLoop:
-        feature_evaluator = self._make_feature_evaluator(input_data)
-        return OuterLoop(
-            self.n_outer,
-            feature_evaluator,
-            self.features_dropout_rate,
-            self.robust_minimum,
+        return outer_loop_results
+
+    def _remove_features(
+        self, features: List[int], results: InnerLoopResults
+    ) -> List[int]:
+        features_to_keep = int(np.floor(len(features) * self.keep_fraction))
+        features = self._select_n_best(results, features_to_keep)
+        return features
+
+    @staticmethod
+    def _select_n_best(inner_loop_result: InnerLoopResults, keep_n: int) -> List[int]:
+        if keep_n < 1:
+            return []
+        ranks = [r.ranks for r in inner_loop_result]
+        avg_ranks = average_ranks(ranks)
+        return get_best_n_features(avg_ranks, keep_n)
+
+    def create_outer_loop_results(
+        self,
+        feature_elimination_results: FeatureEliminationResults,
+        input_data: InputDataset,
+        outer_split: Split,
+        data_splitter: DataSplitter,
+    ) -> OuterLoopResults:
+        min_eval, mid_eval, max_eval = self.evaluate_min_mid_and_max_features(
+            input_data,
+            feature_elimination_results.best_features,
+            outer_split,
+            data_splitter,
+        )
+        outer_loop_results = OuterLoopResults(
+            min_eval=min_eval,
+            mid_eval=mid_eval,
+            max_eval=max_eval,
+            n_features_to_score_map=feature_elimination_results.n_features_to_score_map,
         )
+        return outer_loop_results
+
+    def evaluate_min_mid_and_max_features(
+        self,
+        input_data: InputDataset,
+        best_features: SelectedFeatures,
+        split: Split,
+        data_splitter: DataSplitter,
```

Comment:
Yeahh... this one is weird here ;)

Reply:
what do you mean?

Reply:
That it needs the data_splitter, although it already was given a split is a bit weird imho